### ISSUE TITLE
puppet 4.2 / puppet-agent-1.2 no longer use Ruby-based facter

### ISSUE THREAD

**Issue by gdhbashton** [puppet-community/puppet-mcollective#224]
The cronjob `/opt/puppetlabs/mcollective/plugins/refresh-mcollective-metadata` fails with Puppet 4.2.0 / Facter 3.0.1 because of

```
/opt/puppetlabs/puppet/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- facter/application (LoadError)
	from /opt/puppetlabs/puppet/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /opt/puppetlabs/mcollective/plugins/refresh-mcollective-metadata:4:in `<main>'
```

And quite rightly the path `/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/facter` no longer exists because the rewritten facter version 3 in C no longer requires any of those Ruby files.

My local workaround in the `mcollective` module is to replace the existing cronjob with `facter --yaml >/etc/puppetlabs/mcollective/facts.yaml` every 15 minutes. 

Do you feel this is an appropriate solution for the module if wrapped in a `if facterversion > 3` condition? I'll prepare a PR if so.

Cheers,
Gavin.

**Comment by gdhbashton** [comment]
Submitted PR #225 anyway :)

**Comment by igalic** [comment]
this should be fixed with #228
